src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAroundCheck.java

```diff
@@ -862,14 +879,28 @@ private boolean isColonOfForEach(int parentType) {
     /**
      * Is array initialization.
      *
-     * @param currentType current token
-     * @param parentType parent token
+     * @param currentType current token.
+     * @param parentType parent token.
      * @return true is current token inside array initialization
      */
-    private static boolean isArrayInitialization(int currentType, int parentType) {
-        return currentType == TokenTypes.RCURLY
-                && (parentType == TokenTypes.ARRAY_INIT
-                        || parentType == TokenTypes.ANNOTATION_ARRAY_INIT);
+    private boolean isArrayInitialization(int currentType, int parentType) {
+        return currentType == TokenTypes.RCURLY && parentType == TokenTypes.ANNOTATION_ARRAY_INIT
+                || parentType == TokenTypes.ARRAY_INIT;
+    }
+
+    /**
+     * Is array initialization with consideration of allowWhitespaceAroundArrayInit.
+     *
+     * @param currentType current token.
+     * @param parentType parent token.
+     * @return true is current token inside array initialization and
+     *         allowWhitespaceAroundArrayInit is set to true.
+     */
+    private boolean isArrayInitializationWithProperty(int currentType, int parentType) {
```

Comment:
If I had done-
```
private boolean isArrayInitialization(int currentType, int parentType) {
        return currentType == TokenTypes.RCURLY && parentType == TokenTypes.ANNOTATION_ARRAY_INIT
                || parentType == TokenTypes.ARRAY_INIT && allowWhitespaceAroundArrayInit;
    }
```
There would have been a problem in this line
The problem would be that it will always check `}` for whitespace around even when `allowWhitespaceAroundArrayInit` is `false`
https://github.com/checkstyle/checkstyle/blob/3d8b43445818246cd34ee97b3739dac75b10fae6/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAroundCheck.java#L736
Also, I couldn't call `allowWhitespaceAroundArrayInit` as some logical checkstyle error was there.

Reply:
I had to make another method. If I had decided to make a boolean inside `isNotRelevantSituation`-
```
final boolean = currentType == TokenTypes.RCURLY
                && parentType == TokenTypes.ARRAY_INIT
                || parentType == TokenTypes.ANNOTATION_ARRAY_INIT;
```
I would have faced cyclomatic complexity as number paths would have been 11, max allowed is 10.